### PR TITLE
Add ifconfig for layout updates

### PR DESCRIPTION
**Pull request by luigifab**
### Description

This PR add the possibility to `update handle` with `ifconfig`, like for `action`.

```xml
<something>
    <update handle="someone" ifconfig="x/y/z" />
</something>
```

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Review by fballiano — APPROVED** [review]
If I remember correctly it was part of some module but LGTM and it's a good addition

File: app/code/core/Mage/Core/Model/Layout/Update.php [hunk 1]
```diff
@@ -405,10 +405,16 @@ protected function _getUpdateString($handle)
...
-            if (strtolower($child->getName()) == 'update' && isset($child['handle'])) {
-                $this->merge((string)$child['handle']);
-                // Adding merged layout handle to the list of applied hanles
-                $this->addHandle((string)$child['handle']);
+            if ((strtolower($child->getName()) == 'update') && isset($child['handle'])) {
+                $allow = true;
+                if (isset($child['ifconfig']) && ($configPath = (string)$child['ifconfig'])) {
+                    $allow = Mage::getStoreConfigFlag($configPath);
+                }
+                if ($allow) {
```
Comment (sreichel) [inline app/code/core/Mage/Core/Model/Layout/Update.php:413]:
```suggestion
            if (strtolower($child->getName()) == 'update' && isset($child['handle'])) {
                $allow = true;
                if (isset($child['ifconfig'])) {
                    $allow = Mage::getStoreConfigFlag((string)$child['ifconfig']);
                }
                if ($allow) {
```

Comment (luigifab) [reply app/code/core/Mage/Core/Model/Layout/Update.php:413]: I copy pasted logic from:

https://github.com/OpenMage/magento-lts/blob/412d44f42036ded84f5d7e96312605de3bdc80f0/app/code/core/Mage/Core/Model/Layout.php#L290-L295

**Review by sreichel — APPROVED** [review]

File: app/code/core/Mage/Core/Model/Layout/Update.php (hunk 1 above)
Comment (sreichel) [reply app/code/core/Mage/Core/Model/Layout/Update.php:413]: then we reuse it :+1:

**Comment by fballiano** [comment]
merged and v20ed

**Pull request merged**